numba/core/errors.py

```diff
@@ -544,7 +544,7 @@ class NotDefinedError(IRError):
 
     def __init__(self, name, loc=None):
         self.name = name
-        msg = "Variable '%s' is not defined." % name
+        msg = "The compiler failed to analyze the bytecode"
```

Comment:
Perhaps combine the two messages? It's important to tell the user that the compiler failed to analyse the bytecode, but also useful to say why.

Reply:
That does make sense. I just committed the change you suggested, where I merged the two messages into one.